crates/goose/src/agents/agent.rs

```diff
@@ -781,8 +781,13 @@ impl Agent {
             .ok_or_else(|| anyhow::anyhow!("Session {} has no conversation", session_config.id))?;
 
         let needs_auto_compact = !is_manual_compact
-            && crate::context_mgmt::check_if_compaction_needed(self, &conversation, None, &session)
-                .await?;
+            && crate::context_mgmt::check_if_compaction_needed(
```

Comment:
any particular reason to not have these imported with a `use crate::context_mgmt::{...}`?